File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.size() == 0) return 0;
        vector<int> curr(96, 0);

        int l = 0, r = 0;
        curr[s[0] - ' ']++;
        int answer = 1;

        while (r + 1 < s.size()) {
            r += 1;
            curr[s[r] - ' ']++;
            while (curr[s[r] - ' '] > 1) {
                curr[s[l] - ' ']--;
                l++;
            }

            answer = max(answer, r - l + 1);
        }

        return answer;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        items = Counter(nums)

        answer = 0
        for i in set(nums):
            if items.get(i - 1) is None:
                j = i
                local = 1
                while items.get(j + 1):
                    j = j + 1
                    local += 1
                answer = max(answer, local)
        
        return answer
        